src/main/java/com/github/joschi/jersey/security/smime/EnvelopedReader.java

```diff
@@ -61,14 +60,15 @@ public EnvelopedInput readFrom(Class<EnvelopedInput> type, Type genericType, Ann
         if (headers.containsKey("Content-Type")) {
             headerString.append("Content-Type: ").append(headers.getFirst("Content-Type")).append(CRLF);
         }
-        if (headers.containsKey("Content-Transfer-Encoding")) {
-            headerString.append("Content-Transfer-Encoding: ").append(headers.getFirst("Content-Transfer-Encoding")).append(CRLF);
+        InputStream decodedEntityStream = entityStream;
+        if ("base64".equalsIgnoreCase(headers.getFirst("Content-Transfer-Encoding"))) {
+            decodedEntityStream = java.util.Base64.getMimeDecoder().wrap(entityStream);
```

Comment:
This class is only available in Java SE 8 or later. Since Jersey 2.x is targeting Java SE 7, this project should run with it too. There are several alternative Base64 implementations available in this very project (e. g. the one from BouncyCastle or `com.github.joschi.jersey.util.Base64`).